test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -153,28 +151,34 @@ void performFailoverTest() throws InterruptedException {
             while (!testState.stopWriteFlag.get()) {
                 try {
                     long value = testState.eventData.incrementAndGet();
-                    Thread.sleep(100);
+                    Exceptions.handleInterrupted(() -> Thread.sleep(100));
                     log.debug("Event write count before write call {}", value);
                     writer.writeEvent(String.valueOf(value), value);
                     log.debug("Event write count before flush {}", value);
                     writer.flush();
                     testState.eventWriteCount.getAndIncrement();
                     log.debug("Writing event {}", value);
-                } catch (InterruptedException e) {
-                    log.error("Error in sleep: ", e);
-                    testState.getWriteException.set(e);
-                } catch (ConnectionClosedException e) {
-                    log.warn("Test exception in writing events: ", e);
-                    continue;
                 } catch (Throwable e) {
                     log.error("Test exception in writing events: ", e);
                     testState.getWriteException.set(e);
                 }
             }
-            writer.close();
+            closeWriter(writer);
         }, executorService);
     }
 
+    private void closeWriter(EventStreamWriter<Long> writer) {
+        try {
+            log.info("Closing writer");
+            writer.close();
+        } catch (RetriesExhaustedException e) {
```

Comment:
As we have seen, this is not going to work because the exception is wrapped. We should probably change the exception handling here.